File: src/main/java/com/example/oauth/model/RefreshTokenAttributes.java
package com.example.oauth.model;

import com.example.oauth.repository.model.SocialUser;
import io.jsonwebtoken.Claims;
import lombok.Getter;

@Getter
public class RefreshTokenAttributes extends TokenAttributes {
    private Long id;
    // TODO : discuss token attributes

    private RefreshTokenAttributes() {
        super(TokenType.REFRESH_TOKEN);
    }

    protected static RefreshTokenAttributes of(SocialUser socialUser) {
        RefreshTokenAttributes tokenAttributes = new RefreshTokenAttributes();
        tokenAttributes.id = socialUser.getId();
        return tokenAttributes;
    }

    protected static RefreshTokenAttributes of(Claims claims) {
        RefreshTokenAttributes tokenAttributes = new RefreshTokenAttributes();
        tokenAttributes.id = claims.get("id", Long.class);
        return tokenAttributes;
    }

    @Override
    public SocialUser getPseudoSocialUser() {
        return SocialUser.builder()
                .id(id)
                .build();
    }
}

File: src/main/java/com/example/oauth/model/token/RefreshTokenAttributes.kt
package com.example.oauth.model.token

import com.example.oauth.repository.model.SocialUser
import io.jsonwebtoken.Claims

class RefreshTokenAttributes(
    private var id: Long,
) : TokenAttributes(TokenType.REFRESH_TOKEN) {

    override val pseudoSocialUser = SocialUser(
        id = id
    )

    companion object {
        fun of(socialUser: SocialUser): RefreshTokenAttributes {
            return RefreshTokenAttributes(
                id = socialUser.id!!
            )
        }

        fun of(claims: Claims): RefreshTokenAttributes {
            return RefreshTokenAttributes(
                id = claims.get("id", Long::class.java)
            )
        }
    }

}